sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/AttributeMap.scala

```diff
@@ -23,12 +23,14 @@ package org.apache.spark.sql.catalyst.expressions
  * of the name, or the expected nullability).
  */
 object AttributeMap {
+  val empty = new AttributeMap(Map.empty)
```

Comment:
```Scala
def empty[A]: AttributeMap[A] = new AttributeMap(Map.empty)
```

Reply:
@gatorsmile this was the initial version. then came the idea of @mgaido91 and @maropu to use val (instead of def) either here or in ResolveReferences to spare some empty() calls

Reply:
Reverted to the previous one where we have one val in ResolveReferences.